### ISSUE TITLE
The application exited with an assertion when tried to use Font Awesome Icons.

### ISSUE THREAD

**Issue by vidcentum** [ocornut/imgui#4479]
Hi:

The application exited with an assertion when tried to use Font Awesome Icons (version 5, Not PRO version).

**About Dear ImGUI:**
```
Dear ImGui 1.85 WIP (18408)
--------------------------------
sizeof(size_t): 8, sizeof(ImDrawIdx): 4, sizeof(ImDrawVert): 20
define: __cplusplus=201703
define: __linux__
define: __GNUC__=9
--------------------------------
io.BackendPlatformName: imgui_impl_glfw
io.BackendRendererName: imgui_impl_opengl3
io.ConfigFlags: 0x00000001
 NavEnableKeyboard
io.ConfigInputTextCursorBlink
io.ConfigWindowsResizeFromEdges
io.ConfigMemoryCompactTimer = 60.0
io.BackendFlags: 0x0000000E
 HasMouseCursors
 HasSetMousePos
 RendererHasVtxOffset
--------------------------------
io.Fonts: 1 fonts, Flags: 0x00000000, TexSize: 512,64
io.DisplaySize: 1920.00,968.00
io.DisplayFramebufferScale: 1.00,1.00
--------------------------------
style.WindowPadding: 8.00,8.00
style.WindowBorderSize: 1.00
style.FramePadding: 4.00,3.00
style.FrameRounding: 0.00
style.FrameBorderSize: 0.00
style.ItemSpacing: 8.00,4.00
style.ItemInnerSpacing: 4.00,4.00
```

We are trying to use Font Awesome Icons Version 5 (Not PRO version).  
```
Refer to https://github.com/juliettef/IconFontCppHeaders
```

Followed the steps as outline in the ``` https://github.com/ocornut/imgui/blob/master/docs/FONTS.md ```

Tried with ../fonts/Roboto-Medium.ttf and AddFontDefault() separately. In both the cases, the application exited with an assertion.

**Result:** The application exited with an assertion.

**Error Message:**
imgui/imgui_draw.cpp:2377: bool ImFontAtlasBuildWithStbTruetype(ImFontAtlas*): Assertion `font_offset >= 0 && "FontData is incorrect, or FontNo cannot be found."' failed.

**Code Snippet:**
```
  // io.Fonts->AddFontFromFileTTF("../fonts/Roboto-Medium.ttf", 16.0f); 
  io.Fonts->AddFontDefault();
  
  // merge in icons from Font Awesome
  static const ImWchar icons_ranges[] = { ICON_MIN_FA, ICON_MAX_FA, 0 };
  
  ImFontConfig icons_config;
  icons_config.MergeMode = true;
  icons_config.GlyphMinAdvanceX = 13.0f; // Use if you want to make the icon monospaced
  
  io.Fonts->AddFontFromFileTTF( "../fonts/fa-solid-900.ttf", 13.0f, &icons_config, icons_ranges );
  // The ../fonts/fa-solid-900.ttf file is available in the path.
```

Could you please help where we are missing while using Font Awesome Icons?

Thank you in advance.

Best Regards
Maruthi

**Comment by ocornut** [comment]
As stated by the error you posted, I believe your font file is incorrect and not a valid TTF file. Try to open it in a font viewer (such as Windows’s one).

**Comment by vidcentum** [comment]
Thank you, @ocornut . The issue was with bad TTF file. I downloaded again and opened first in a viewer. Now I am able to use Font Awesome Icons. Thank you once again.

Task: Downloaded TTF files again and placed them in fonts/ 

Test 1:  Merged Roboto-Medium.ttf with Font Awesome 
Result: Success

Test 2: Merged default AddFontDefault() with Font Awesome
Result: Success

Reason for the issue: Bad TTF file was used. 

I am closing the issue. 

Thank you very much. 

Best Regards
Maruthi
